qemu/tests/virtio_fs_share_data.py

```diff
@@ -455,7 +455,23 @@ def viofs_svc_stop_start(action, cmd, expect_status):
                     session.cmd(cmd_make % fs_dest, io_timeout)
                     status, output = session.cmd_status_output(
                         cmd_pjdfstest % fs_dest, io_timeout)
-                    if status != 0:
+                    failed_test = output.split("-------------------")[1]\
+                                        .split("Files=")[0]
+                    # ignore the specific failed cases from pjdfstest
+                    ignore_cases = params.objects("pjdfstest_blacklist")
+                    cases_in_output = list(re.findall(r"/mnt.*\.t", failed_test))
+                    false_in_list = [False for _ in range(len(cases_in_output))]
+                    cases_in_output = dict(zip(cases_in_output, false_in_list))
+                    for case in cases_in_output.keys():
+                        for ig_case in ignore_cases:
+                            if ig_case in case:
+                                error_context.context("Warn: %s was failed!"
+                                                      % ig_case, test.log.debug)
+                                cases_in_output[case] = True
+
+                    unexpected_fail_case = list(cases_in_output.values()).count(False)
+
+                    if status != 0 and unexpected_fail_case > 0:
```

Comment:
For the part of handling to filter the expected fail case, I think could we remove the corresponding case scirpt before make and make install pjdfstest instead of this part of codes?
For example, to remove the `pjdfstest/tests/rename/21.t` and `pjdfstest/tests/symlink/03.t`, we can have a try it.

Reply:
> For the part of handling to filter the expected fail case, I think could we remove the corresponding case scirpt before make and make install pjdfstest instead of this part of codes? For example, to remove the `pjdfstest/tests/rename/21.t` and `pjdfstest/tests/symlink/03.t`, we can have a try it.

Hello @xiagao , what's your opinion? The way "filter failed case" is already enough for this case.

Reply:
Hi @nickzhq I think the current solution is ok for me.

Reply:
@YongxueHong hello sorry for the late reply,
Actually I think the current logic is more suitable for me, we will test all the cases and manually choose the ones to ignore. In this way, when the subsequent version of the case passes, we can quickly perceive it. 

```
[stdlog] 2022-09-12 22:26:15,722 aexpect.client DEBUG| Sending command: prove -rv /mnt/myfs/pjdfstest/tests
[stdlog] 2022-09-12 22:29:34,097 aexpect.client DEBUG| Sending command: echo $?
[stdlog] 2022-09-12 22:29:34,206 avocado.test DEBUG| Context: Warn: pjdfstest/tests/rename/21.t was failed!
[stdlog] 2022-09-12 22:29:34,206 avocado.test DEBUG| Context: Warn: pjdfstest/tests/symlink/03.t was failed!
```

Reply:
> @YongxueHong hello sorry for the late reply, Actually I think the current logic is more suitable for me, we will test all the cases and manually choose the ones to ignore. In this way, when the subsequent version of the case passes, we can quickly perceive it.
> 
> ```
> [stdlog] 2022-09-12 22:26:15,722 aexpect.client DEBUG| Sending command: prove -rv /mnt/myfs/pjdfstest/tests
> [stdlog] 2022-09-12 22:29:34,097 aexpect.client DEBUG| Sending command: echo $?
> [stdlog] 2022-09-12 22:29:34,206 avocado.test DEBUG| Context: Warn: pjdfstest/tests/rename/21.t was failed!
> [stdlog] 2022-09-12 22:29:34,206 avocado.test DEBUG| Context: Warn: pjdfstest/tests/symlink/03.t was failed!
> ```

Hi @zhenyzha 
Yes, your concern makes sense. Okay, let's keep the current design.